public/app/features/dimensions/editors/ValueMappingsEditor/ValueMappingsEditor.tsx

```diff
@@ -1,16 +1,20 @@
 import React, { useCallback, useMemo, useState } from 'react';
-import { GrafanaTheme2, MappingType, ValueMapping } from '@grafana/data';
+import { FieldConfigProperty, GrafanaTheme2, MappingType, StandardEditorProps, ValueMapping } from '@grafana/data';
 import { css } from '@emotion/css';
 import { buildEditRowModels, editModelToSaveModel, ValueMappingsEditorModal } from './ValueMappingsEditorModal';
 import { useStyles2, VerticalGroup, Icon, ColorPicker, Button, Modal } from '@grafana/ui';
+import { MediaType, ResourceFolderName, ResourcePickerSize } from '../../types';
+import { ResourcePicker } from '../ResourcePicker';
 
-export interface Props {
-  value: ValueMapping[];
-  onChange: (valueMappings: ValueMapping[]) => void;
+export interface Props extends StandardEditorProps<ValueMapping[], any, any> {
+  showIcon?: boolean;
 }
 
-export const ValueMappingsEditor = React.memo(({ value, onChange }: Props) => {
+export const ValueMappingsEditor = React.memo((props: Props) => {
+  const { value, onChange, item } = props;
+
   const styles = useStyles2(getStyles);
+  const showIconPicker = item.settings?.enableUncommonOptions?.includes(FieldConfigProperty.Icon);
```

Comment:
These things seem not 100% correlated, just because a panel support icon on the field level does not mean you support it on the value level. Think it might be easier to just have a settings option for ValueMappings that says it supports icons. Like we have for color options: https://github.com/grafana/grafana/blob/f2f810599c616793d39faac84e5ab2525fbcd0a2/public/app/plugins/panel/timeseries/config.ts#L47